apps/aechannel/src/aesc_snapshot_solo_tx.erl

```diff
@@ -0,0 +1,180 @@
+%%%=============================================================================
+%%% @copyright 2018, Aeternity Anstalt
+%%% @doc
+%%%    Module defining the State Channel solo snapshot transaction
+%%% @end
+%%%=============================================================================
+-module(aesc_snapshot_solo_tx).
+
+-behavior(aetx).
+
+%% Behavior API
+-export([new/1,
+         type/0,
+         fee/1,
+         ttl/1,
+         nonce/1,
+         origin/1,
+         check/5,
+         process/5,
+         signers/2,
+         serialization_template/1,
+         serialize/1,
+         deserialize/2,
+         for_client/1
+        ]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(CHANNEL_SNAPSHOT_SOLO_TX_VSN, 1).
+-define(CHANNEL_SNAPSHOT_SOLO_TX_TYPE, channel_snapshot_solo_tx).
+-define(CHANNEL_SNAPSHOT_SOLO_TX_FEE, 4).
+
+-type vsn() :: non_neg_integer().
+
+-record(channel_snapshot_solo_tx, {
+          channel_id :: aec_id:id(),
+          from       :: aec_id:id(),
+          payload    :: binary(),
+          ttl        :: aetx:tx_ttl(),
+          fee        :: non_neg_integer(),
+          nonce      :: non_neg_integer()
+         }).
+
+-opaque tx() :: #channel_snapshot_solo_tx{}.
+
+-export_type([tx/0]).
+
+%%%===================================================================
+%%% Behaviour API
+%%%===================================================================
+
+-spec new(map()) -> {ok, aetx:tx()}.
+new(#{channel_id := ChannelIdBin,
+      from       := FromPubKey,
+      payload    := Payload,
+      fee        := Fee,
+      nonce      := Nonce} = Args) ->
+    Tx = #channel_snapshot_solo_tx{
+            channel_id = aec_id:create(channel, ChannelIdBin),
```

Comment:
This doesn't match the API where `aec_id:id()` is passed into the transaction.

Please rebase on master and amend this.